io-engine/src/bdev/nexus/nexus_child.rs

```diff
@@ -1022,7 +1089,19 @@ impl<'c> NexusChild<'c> {
         }
     }
 
-    /// Destroys the child's block device.
+    /// Destroys the child's underlying block device.
+    ///
+    /// While the desvice is being destroyed, the child will temporarily stay in
+    /// `Destroying` state.
+    ///
+    /// Once the device is destroyed, the child transits
+    /// to `Closed` state or reverts to the state before the device destroy:
+    ///
+    /// * If the child was `Open` before destroying the device, it will transit
+    ///   to `Closed`.
+    ///
+    /// * If the child was `Faulted` before destroying the device, it will
+    ///   revert back to `Faulted`.
```

Comment:
:+1:

Reply:
"Retire" is a Solaris term. Solaris has a Fault Management (FM) system which keeps track of all H/W issues (and some S/W), including disks, and allows to administrate all that in a centralized way. "Retired" means a disk device that was faulted, and the system won't try to access it (e.g. on boot). A retired device can be "unretired" by an admin when the problem is fixed (e.g. like cable reconnected). Or an admin may decide to replace a retired device.

In io-engine, "retire" term is not quite the same as in Solaris, although related. It is used for the multi-step process of shutting down a faulted device: disconnect it from I/O channels, notify listeners, shutdown related NVMF stuff, etc, etc, and eventually close the Bdev and release the underlying Linux device.